[TEQ-26] Checking devDependencies.

philtuskha opened the pull request:
And failing when packages are listed there, explaining that they should be listed under dependencies instead.

philtuskha committed:
checking devDependencies.

jbradberry reviewed (changes requested):
Per our conversation, this should
- check both dependencies and devDependencies
- issue a warning instead of an error
- only in the case where deps is empty and devdeps is not

Additionally, we are starting to get enough js tasks going, I'd like for all except the initial check for packages.json to be broken out into its own file which is conditionally run, similar to the web.yml and worker.yml includes at the bottom of tasks/main.yml.

philtuskha committed:
changed unforgiving fail to casual warning

philtuskha committed:
added name to debug

jbradberry reviewed (changes requested):
Could you please break out the javascript-y tasks into their own yml file that gets conditionally included?  That includes your two, plus the check for build script task above them.

Also, please bring the CHANGES.rst file up to date.

jbradberry commented:
I'm going to take back my request to break out the npm tasks.

jbradberry committed:
Update CHANGES.rst

to reflect that this change no longer stops the deployment, it just issues a message.
